scylla/src/transport/load_balancing/default.rs

```diff
@@ -605,21 +605,21 @@ mod tests {
         };
 
         enum ExpectedGroup {
-            AnyOrder(HashSet<u16>),
+            NonDeterministic(HashSet<u16>),
             Deterministic(HashSet<u16>),
         }
```

Comment:
So now we expect nondeterminism both for replicas and for non-replicas, which is a generalisation. Replicas are shuffled, whereas non-replicas are randomly rotated. If we want to consider both as nondeterministic, *dot*, then we can leave it as you propose; if, however, the distinction between them is meaningful, we should highlight that distinction at least in naming.

Reply:
I'd rather leave this as "non-deterministic". If we wanted to differentiate "shuffling" and "rotating" then we probably could provide different assertions for both types. I'd rather not do it in this PR.